plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -164,6 +164,27 @@ fu_plugin_dell_dock_get_ec (GPtrArray *devices)
 	return ec_parent;
 }
 
+gboolean
+fu_plugin_update (FuPlugin *plugin,
```

Comment:
maybe fu_plugin_runner_composite_prepare instead?

Reply:
Actually, maybe this is a hack, as it seems the "last set" thing wins. Shouldn't we get the metadata from the FuDevice directly? Maybe an extra vfunc or property? I'm happy to do the plumbing if you think this is a good idea.

Reply:
Sure a vfunc on the device that adds metadata sounds good. Go ahead :)

Reply:
Maybe we want everything in `fu_device_set_metadata()`? I can't see anything that we wouldn't want to include.

Reply:
Ya, I *think* doing `fu_device_set_metadata(dev, "DellDockSKU", sku)` gives us the metadata in in the report right now.

Reply:
We might need to be cognizant that there is no PII in the metadata, but otherwise yes I think that makes sense.

Reply:
I tried `fu_device_set_metadata*` and it doesn't flow into the report.  However I do like doing it in `composite_prepare` as this metadata can then be set even in the context that another non-dell-dock plugin performs the update.

Think of it as "If this is a Dell dock add the metadata in the upload report of which model it was"